umapi_client/functional.py

```diff
@@ -111,31 +82,13 @@ def __init__(self, id_type=IdentityTypes.adobeID, email=None, username=None, dom
             raise ArgumentError("Identity type (%s) must be one of %s" % (id_type, [i.name for i in IdentityTypes]))
         self.id_type = id_type
         self.email = None
-        self.domain = None
-        if username:
-            if email and username.lower() == email.lower():
-                # ignore the username if it's the same as the email (policy default)
-                username = None
-            elif id_type is not IdentityTypes.federatedID:
```

Comment:
I don't see why you are removing this check.  You can't specify username at all unless it's a Federated ID.  I can see rewording the error message to make that clearer.